enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  ## my_all method
+  def my_all
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_none method
+  def my_none
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_count method
+  def my_count
+    return self unless block_given?
+
+    arr = self
+    counter = 0
+    arr.my_each { |i| counter += 1 if yield(i) }
+    counter
+  end
+
+  ## my_map method
+  def my_map(&factor)
+    return self unless block_given?
+
+    arr = self
+    new_arr = []
+    arr.my_each { |x| new_arr << factor.call(x) }
+    new_arr
+  end
+
+  ## my_inject method
+  def my_inject(&factor)
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |x| factor.call(x) }
+  end
```

Comment:
- [ ] enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
- [ ] enumerables #my_inject searches for the longest word in an array of strings
      Failure/Error: let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }
- [ ] enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: let(:operation) { proc { |sum, n| sum + n } }
- [ ] enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range
      Failure/Error: actual = range.my_inject { |prod, n| prod * n }
- [ ] enumerables #my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
- [ ] enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
- [ ] enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)

- [ ] enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)

- [ ] enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range
      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)